packages/@aws-cdk/aws-pipes-alpha/lib/enrichment.ts

```diff
@@ -0,0 +1,112 @@
+   */
+export interface EnrichmentParametersProperty {
+  /**
+   * Contains the HTTP parameters to use when the target is a API Gateway REST endpoint or EventBridge ApiDestination.
+   *
+   * If you specify an API Gateway REST API or EventBridge ApiDestination as a target, you can use this parameter to specify headers, path parameters, and query string keys/values as part of your target invoking request. If you're using ApiDestinations, the corresponding Connection can also have these values configured. In case of any conflicting keys, values from the Connection take precedence.
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipeenrichmentparameters.html#cfn-pipes-pipe-pipeenrichmentparameters-httpparameters
+   * @default - none
+   */
+  readonly httpParameters?: EnrichmentHttpParametersProperty;
+
+  /**
+   * Valid JSON text passed to the enrichment.
+   *
+   * In this case, nothing from the event itself is passed to the enrichment. For more information, see [The JavaScript Object Notation (JSON) Data Interchange Format](https://docs.aws.amazon.com/http://www.rfc-editor.org/rfc/rfc7159.txt) .
+   *
+   * To remove an input template, specify an empty string.
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipeenrichmentparameters.html#cfn-pipes-pipe-pipeenrichmentparameters-inputtemplate
+   * @default - none
+   */
+  readonly inputTransformation?: IInputTransformation;
+}
+
+/**
+ * Enrichment step to enhance the data from the source before sending it to the target.
+ */
+export interface IEnrichment {
+  /**
+   * The ARN of the enrichment resource.
+   *
+   * Length Constraints: Minimum length of 0. Maximum length of 1600.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-pipes-pipe.html#cfn-pipes-pipe-enrichment
+   */
+  readonly enrichmentArn: string;
+
+  /**
+   * The parameters required to set up enrichment on your pipe.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-pipes-pipe.html#cfn-pipes-pipe-enrichmentparameters
+   */
+  readonly enrichmentParameters: EnrichmentParametersProperty;
+
+  /**
+   * Grant the pipes role to invoke the enrichment.
+   */
+  grantInvoke(grantee: IRole): void;
+}
+
+/**
+ * Enrichment step to enhance the data from the source before sending it to the target.
+ *
+ * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/pipes-enrichment.html
+ */
+export abstract class Enrichment implements IEnrichment {
+  public readonly enrichmentArn: string;
+  public readonly enrichmentParameters: EnrichmentParametersProperty;
+
+  constructor(
+    enrichmentArn: string,
+    props: EnrichmentParametersProperty,
+  ) {
+    this.enrichmentParameters = props;
+    // TODO - validate ARN is a valid enrichment ARN based on regex from cfn
```

Comment:
Please remove todo. Either just implement this or create an issue in the backlog with a label for the module. Before stabilization we will go through all issue in the backlog.